theblues/charmstore.py

```diff
@@ -0,0 +1,308 @@
+import logging
+
+import requests
+from requests.exceptions import HTTPError
+
+from errors import (
+    EntityNotFound,
+    ServerError,
+    )
+
+try:
+    from urllib import quote
+except:
+    from urllib.parse import quote
+
+
+class CharmStore(object):
+    """A connection the charmstore."""
+
+    def __init__(self, url, macaroons=None, verify=True):
+        super(CharmStore, self).__init__()
+        self.url = url
+        self.verify = verify
+        self.macaroons = macaroons
+
+    def _get(self, url):
+        cookies = dict([('macaroon-storefront', self.macaroons)])
+        try:
+            response = requests.get(url, verify=self.verify, cookies=cookies)
+            response.raise_for_status()
+            return response
+        # XXX: To be reviewed when splitting the library.
+        # Is it te right place to log or should we let the users of the blues
+        # to handle logging ?
```

Comment:
It is, but not in this branch; I'll put a card up in weekly retrospective to discuss this one, and if there's work to be done we'll add it.